sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects/objects.scala

```diff
@@ -678,6 +678,9 @@ object MapObjects {
       elementType: DataType,
       elementNullable: Boolean = true,
       customCollectionCls: Option[Class[_]] = None): MapObjects = {
+    if (function == null) {
+      throw new UnsupportedOperationException("Cannot instantiate MapObjects with null function")
+    }
```

Comment:
assert or reuqire?

Reply:
btw, could we remove the workaround now? https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects/objects.scala#L653
Or, we need to update the comment if the same issue can happen in the other places.

Reply:
> could we remove the workaround now?

IIUC, the original reason for that was some kind of graceful failure if someone coded an Aggregator wrong, which I assume could still happen.

In retrospect, what would have helped me somewhat was either a comment in the exception message above, or at least code comment, saying something like "this exception is likely happening because an UnresolvedMapObject was serialized without being resolved".

Reply:
The particular reason this happened to me should no longer be possible. I'm not aware of any other path for it to happen, but leaving an informative exception message, or at least a code comment to annotate the non-null assertion, seems like good defensive programming.

Reply:
But, This part `(e.g. mistakenly reference an encoder instance when implementing
 * Aggregator)`  will be fixed by this PR, right?